modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java

```diff
@@ -419,7 +420,8 @@ public CodegenDiscriminator getDiscriminator() {
 
     public void setDiscriminator(CodegenDiscriminator discriminator) {
         this.discriminator = discriminator;
-        if (discriminator != null && !discriminator.getMappedModels().isEmpty()) {
+        if (discriminator != null && (!discriminator.getMappedModels().isEmpty()) || this.vendorExtensions.containsKey("x-discriminator-value")) {
```

Comment:
Thanks for the PR!

The 2nd condition on this line is surrounded by unnecessary parenthesis. So I want to check - did you mean to write `if (discriminator != null && (!discriminator.getMappedModels().isEmpty() || this.vendorExtensions.containsKey("x-discriminator-value"))) {` (with parenthesis around the 2nd and 3rd condition) instead?
If it's not a mistake and you simply added the parenthesis for clarity, that's fine for me.

Reply:
@martin-mfg  
Thanks for the review.

**Clarity**.  only processing if there is a discriminator.

In the context of the method:
if x-discriminator-value is defined, discriminator is not null.  So indeed, the parenthesis are not strictly necessary.